airbyte-integrations/connectors/source-netsuite/source_netsuite/streams.py

```diff
@@ -0,0 +1,190 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC
+from collections import OrderedDict
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional
+
+# Basic full refresh stream
+import requests
+import zeep
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.http import HttpStream
+from netsuitesdk import NetSuiteConnection
+from netsuitesdk.api.base import ApiBase
+
+
+class NetsuiteSoapStream(HttpStream, ApiBase, ABC):
+    def __init__(self, nc: NetSuiteConnection, name: str, config: dict, concurrency_limit: int = 1):
+        self.config = config
+        super().__init__()
+        self.obj_name = name
+        self.nc = nc
+
+        self.concurrency_limit = concurrency_limit
+
+        self.entities_will_have_array_structure = ["VendorBill", "JournalEntry", "ExpenseReport", "ExpenseCategories"]
+
+    url_base = "/"
+
+    @staticmethod
+    def as_serialized(data) -> Iterable[Mapping[str, Any]]:
+        return [zeep.helpers.serialize_object(_) for _ in data]
+
+    def is_entities_or_dicts_inside_list(self, data: list) -> bool:
+        first_element = data[0]
+        if isinstance(first_element, OrderedDict):
+            return False
+        elif isinstance(first_element, list):
+            self.is_entities_or_dicts_inside_list(first_element)
+        return True
+
+    def compromise_structure(self, class_object: list, type_name: str) -> object:
+        if (
+            class_object[0].__class__.__name__ in self.entities_will_have_array_structure
+            or type_name in self.entities_will_have_array_structure
+            or not self.is_entities_or_dicts_inside_list(class_object)
+        ):
+            class_object = [class_object]
+
+        return class_object
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return self.obj_name
+
+    def post(self, data) -> OrderedDict:
```

Comment:
what is this method designed for?

Reply:
Stream was not runs with abstract methods, so this method defined